client/src/pages/Bookings/Bookings.tsx

```diff
@@ -0,0 +1,147 @@
+import { useState, useEffect } from 'react';
+import Grid from '@material-ui/core/Grid';
+import CssBaseline from '@material-ui/core/CssBaseline';
+import useStyles from './useStyles';
+import Booking from './Booking/Booking';
+import { CircularProgress, Paper, Typography } from '@material-ui/core';
+import { Request } from '../../interface/Bookings';
+import BookingCalendar from './Calendar/Calendar';
+import { getRequests } from '../../helpers/APICalls/bookings';
+import moment from 'moment';
+
+export default function Bookings(): JSX.Element {
+  const classes = useStyles();
+  const [bookings, setBookings] = useState<Array<Request>>([]);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [nextBooking, setNextBooking] = useState<Request | undefined>();
+  const [currentBookings, setCurrentBookings] = useState<Array<Request>>([]);
+  const [pastBookings, setPastBookings] = useState<Array<Request>>([]);
+  const [selectedDate, setSelectedDate] = useState<Date>(new Date());
+  const [selectedDateBookings, setSelectedDateBookings] = useState<Array<Request>>([]);
+  const today = new Date();
+
+  useEffect(() => {
+    setLoading(true);
+    getRequests().then((data) => {
+      if (data.requests) setBookings(data.requests);
+    });
+    setLoading(false);
```

Comment:
setLoading from true to false renders immediately one after the other, was this meant to be in a callback?